### PR TITLE
[MINT-2501] Provider physician List Modal

### PR DESCRIPTION
**Pull request by garyjzhao**
# MINT-2501

## Description

- Created `ProviderAndSupplierModal`
- Add translations for said modal
- Add button to open up modal in the Participant and Provider page 
- 

## How to test this change

1.  Navigate to Participant and Provider page 1 and add `Medicare providers/suppliers` to the first question
2. Verify that link is there
3. Click link and verify that modal opens 
4. Verify that modal functions correctly/as expected
5. Test the side nav
6. Test filter
7. test pagination
8. verify all matches figma

## PR Author Checklist

- [X] I have provided a detailed description of the changes in this PR.
- [X] I have provided clear instructions on how to test the changes in this PR.
- [X] I have updated tests or written new tests as appropriate in this PR.
- [ ] Updated the [Postman Collection](../MINT.postman_collection.json) if necessary.

## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.

### CODE REVIEW

**Review by ClayBenson94 — COMMENTED** [review]
Changes look good locally!

I was able to find one change that might be worth making - is there any way we could clear the search when the `type` of table changes (like when switching between side-nav items?)

Kapture 2024-08-06 at 14 51 55


If that would require any kind of large refactor I'm fine to approve as-is, but thought I'd ask!

File: src/components/TablePagination/index.scss [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#main-content .usa-pagination {
+.usa-pagination {
     justify-content: left;
 
     &__page-no {
```
Comment (patrickseguraoddball) [inline src/components/TablePagination/index.scss:4]: You have to be really careful about removing styles blanketed across a component.  This is a side affect on another table: 

Screenshot 2024-08-07 at 10 01 00 AM

File: src/views/ModelPlan/ReadOnly/_components/Sidenav/index.tsx [hunk 2]
```diff
@@ -9,7 +9,7 @@ import './index.scss';
 
 interface SideNavProps {
   subComponents: subComponentsProps;
-  isHelpArticle: boolean | undefined;
+  isHelpArticle?: boolean;
```
Comment (patrickseguraoddball) [inline src/views/ModelPlan/ReadOnly/_components/Sidenav/index.tsx:12]: Why was this changed?  Not saying it's wrong, just not sure why it was made

File: src/views/ModelPlan/TaskList/ParticipantsAndProviders/_component/ProviderAndSupplierModal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import ReactModal from 'react-modal';
+import {
+  Button,
+  Grid,
+  GridContainer,
+  Icon,
+  SideNav as TrussSideNav
+  // Table as UswdsTable
```
Comment (patrickseguraoddball) [inline src/views/ModelPlan/TaskList/ParticipantsAndProviders/_component/ProviderAndSupplierModal/index.tsx:10]: Unneeded commented out code

File: src/views/ModelPlan/TaskList/ParticipantsAndProviders/_component/ProviderAndSupplierModal/index.tsx [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+          className={`sidenav-button line-height-sans-3 width-full ${
+            activeType === key ? 'current' : ''
+          }`}
+          onClick={() => setActiveType(key as ExistingProviderSupplierTypes)}
+        >
+          {value}
+        </Button>
+      );
+    }
+  );
```
Comment (patrickseguraoddball) [inline src/views/ModelPlan/TaskList/ParticipantsAndProviders/_component/ProviderAndSupplierModal/index.tsx:61]: You can use `geyKeys` here to preserve the type and not need the type assertion `key as ExistingProviderSupplierTypes` I believe.

File: src/views/ModelPlan/TaskList/ParticipantsAndProviders/_component/ProviderAndSupplierModal/index.tsx [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+        appElement={document.getElementById('root')! as HTMLElement}
+      >
+        <div data-testid="provider-physician-type-modal">
+          <div className="mint-discussions__x-button-container display-flex text-base flex-align-center">
+            <button
+              type="button"
+              data-testid="close-discussions"
+              className="mint-discussions__x-button margin-right-2"
+              aria-label="Close Modal"
+              onClick={closeModal}
```
Comment (patrickseguraoddball) [inline src/views/ModelPlan/TaskList/ParticipantsAndProviders/_component/ProviderAndSupplierModal/index.tsx:80]: This has copied over/unused styles and data-testid from discussion center

File: src/views/ModelPlan/TaskList/ParticipantsAndProviders/_component/ProviderAndSupplierTable/index.tsx [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+        canNextPage={canNextPage}
+        pageIndex={state.pageIndex}
+        pageOptions={pageOptions}
+        canPreviousPage={canPreviousPage}
+        pageCount={pageCount}
+        pageSize={state.pageSize}
+        setPageSize={setPageSize}
+        page={[]}
+      />
+    </>
```
Comment (patrickseguraoddball) [inline src/views/ModelPlan/TaskList/ParticipantsAndProviders/_component/ProviderAndSupplierTable/index.tsx:129]: I have a ticket to capture all the conversions of table pagination [MINT-3034](https://jiraent.cms.gov/browse/MINT-3034).  With the latest upgrade to Truss package we now have access to their pagination.  In my last PR I created a hook to merge the component with functionality.  We should standardize all pagination now.  This won't block the PR but just wanted to note this for future visibility.  The mentioned ticket will rewrite this when worked on.  The hook is called `usePagination` and can be referenced by the Models Approaching Clearance PR that was jsut merged

**Review by patrickseguraoddball — CHANGES_REQUESTED** [review]
There doesn't seem to be any navigation elements visible at certain sizes

Screenshot 2024-08-07 at 9 59 06 AM

**Comment by patrickseguraoddball** [comment]
Also a weird overlap here as certain widths, nothing huge

Screenshot 2024-08-07 at 10 13 32 AM

File: src/views/ModelPlan/TaskList/ParticipantsAndProviders/_component/ProviderAndSupplierTable/index.tsx [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+        tableName={modalT('modal.table.title')}
+        className="margin-bottom-4 maxw-none width-mobile"
+      />
+      <TableResults
+        globalFilter={state.globalFilter}
+        pageIndex={state.pageIndex}
+        pageSize={state.pageSize}
+        filteredRowLength={rows.length}
+        rowLength={data.length}
+      />
```
[the remark this answers is not in the document]
Comment (patrickseguraoddball) [reply src/views/ModelPlan/TaskList/ParticipantsAndProviders/_component/ProviderAndSupplierTable/index.tsx:71]: I think should show `31-38 of 38 results`

**Comment by patrickseguraoddball** [comment]
Spacing here seems a bit tight compared to Figma

Screenshot 2024-08-07 at 10 16 37 AM

**Comment by garyjzhao** [comment]
@ClayBenson94 

https://github.com/user-attachments/assets/3732b41f-27e2-47f3-b0c0-7d04eed66444

File: src/components/TableFilter/index.tsx [hunk 8]
```diff
@@ -50,7 +50,7 @@ const GlobalClientFilter = ({
           // Currently only client-side filtering - updates search filter onChange
           setGlobalFilter(e.target.value);
         }}
-        value={globalFilter}
+        value={globalFilter ?? ''}
```
Comment (garyjzhao) [inline src/components/TableFilter/index.tsx:53]: @ClayBenson94 this one liner addressed what you were asking for 🤣


https://github.com/user-attachments/assets/25f02191-46ab-467d-aa4c-934cfa3c6e10

**Review by patrickseguraoddball — CHANGES_REQUESTED** [review]

**Comment by garyjzhao** [comment]
@patrickseguraoddball, i just confirmed with Natalie that we can push this to PROD once the work is completed. She will review the feature in PROD.

**Review by patrickseguraoddball — CHANGES_REQUESTED** [review]
There's some weirdness with the navigation at different sizes.  It goes to select box, then back to the first side nav

Screenshot 2024-08-08 at 2 03 28 PM

Screenshot 2024-08-08 at 2 02 36 PM

**Review by patrickseguraoddball — APPROVED** [review]

**Pull request merged**